packages/vite/src/node/config.ts

```diff
@@ -1089,6 +1075,27 @@ async function loadConfigFromBundledFile(
   }
 }
 
+async function mergeHookConfig(
```

Comment:
The refactor LGTM. I think the function could be called `runConfigHook` instead so it is more clear that the hooks are being executed.

Reply:
Done!